bundles/io/org.eclipse.smarthome.io.transport.serial/src/main/java/org/eclipse/smarthome/io/transport/serial/SerialPortEvent.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Interface for a serial port event.
+ *
+ * @author Markus Rathgeb - Initial contribution
+ */
+@NonNullByDefault
+public interface SerialPortEvent {
```

Comment:
Hi @amitjoy,
the intention of this serial transport has been to provide a thin abstraction layer for the specific implementations. There is the [Java Communications API] and that API is using the package `javax.comm`. There has also been [RXTX] that provides (v2.1 upwards) that but using another package only (`gnu.io`).
The most implementations I know about support that official API under one of that packages.
To keep the abstraction layer thin I would like to keep that numbers as in the "usual" implementations.
Sure, we could create an enum and assign an internal number to that enumerations but then we need to translate from enum to int on every call and from int to enum on every return where that values will be used (e.g. method that will be added on request).

That is the background...
Should we really add the overhead at runtime just to have enums instead of int and to be strictly speaking just to have some stronger type constraints?

[Java Communications API]: http://www.oracle.com/technetwork/java/index-jsp-141752.html
[RXTX]: https://en.wikibooks.org/wiki/Serial_Programming/Serial_Java#RxTx_2

Reply:
From the coding perspective, it would be more readable to have an enum instead. But I feel, the design perspective is different in this scenario. Yes, you are right. It would be a bit of overhead since it will always be a runtime dynamic execution. You can leave it as it is.